Description: Let ejabberdctl handle more shell metacharacters
 Backport of upstream patch, without it, ejabberdctl can't handle shell
 metacharacters in passwords. Closes: #792831
Author: Holger Weiss <holger@zedat.fu-berlin.de>

Index: ejabberd/ejabberdctl.template
===================================================================
--- ejabberd.orig/ejabberdctl.template
+++ ejabberd/ejabberdctl.template
@@ -405,7 +405,7 @@ ctl()
 ctlexec()
 {
     CONN_NAME=$1; shift
-    COMMAND=$(echo $@ | sed 's/;/\\;/g')
+    COMMAND=$(echo $@ | sed 's/["&$;\|<>()]/\\&/g')
     $EXEC_CMD "$ERL \
       $NAME ${CONN_NAME} \
       -noinput \
